kevm-pyk/src/kevm_pyk/cli.py

```diff
@@ -34,6 +54,269 @@ def node_id_like(s: str) -> NodeIdLike:
+    run_args.add_argument(
+        '--output',
+        type=KRunOutput,
+        choices=list(KRunOutput),
+    )
+    run_args.add_argument(
+        '--expand-macros',
+        dest='expand_macros',
+        default=None,
+        action='store_true',
+        help='Expand macros on the input term before execution.',
+    )
+    run_args.add_argument(
+        '--no-expand-macros',
+        dest='expand_macros',
+        action='store_false',
+        help='Do not expand macros on the input term before execution.',
+    )
+    run_args.add_argument(
+        '--debugger',
+        dest='debugger',
+        action='store_true',
+        help='Run GDB debugger for execution.',
+    )
+
+    kast_args = command_parser.add_parser(
+        'kast',
+        help='Run KEVM program.',
+        parents=[
+            kevm_cli_args.logging_args,
+            kevm_cli_args.target_args,
+            kevm_cli_args.evm_chain_args,
+            kevm_cli_args.k_args,
+            config_args.config_args,
+        ],
+    )
+    kast_args.add_argument('input_file', type=file_path, help='Path to input file.')
+    kast_args.add_argument(
+        '--output',
+        type=PrintOutput,
+        choices=list(PrintOutput),
+    )
+
+    return parser
+
+
+def parse_toml_args(args: Namespace) -> dict[str, Any]:
+    def get_profile(toml_profile: dict[str, Any], profile_list: list[str]) -> dict[str, Any]:
+        if len(profile_list) == 0 or profile_list[0] not in toml_profile:
+            return {k: v for k, v in toml_profile.items() if type(v) is not dict}
+        elif len(profile_list) == 1:
+            return {k: v for k, v in toml_profile[profile_list[0]].items() if type(v) is not dict}
+        return get_profile(toml_profile[profile_list[0]], profile_list[1:])
+
+    toml_args: dict[str, Any] = {}
+    if not hasattr(args, 'config_file') or not args.config_file.is_file():
+        return {}
+
+    with open(args.config_file, 'rb') as config_file:
+        try:
+            toml_args = tomli.load(config_file)
+        except tomli.TOMLDecodeError:
+            _LOGGER.error(
+                'Input config file is not in TOML format, ignoring the file and carrying on with the provided command line agruments'
+            )
+
+    toml_args = (
+        get_profile(toml_args[args.command], args.config_profile.split('.')) if args.command in toml_args else {}
+    )
+
+    toml_adj_args: dict[str, Any] = {}
+    for k, v in toml_args.items():
+        opt_string = get_option_string_destination(args.command, k)
+        if opt_string[:3] == 'no_':
+            toml_adj_args[opt_string[3:]] = not v
+        elif k == 'optimization-level':
+            level = toml_args[k] if toml_args[k] >= 0 else 0
+            level = level if toml_args[k] <= 3 else 3
+            toml_adj_args['o' + str(level)] = True
+        else:
+            toml_adj_args[opt_string] = v
+
+    return toml_adj_args
```

Comment:
This function is very similar to the one already in pyk. Is it possible to use that one instead and then to do any KEVM-specific adjustments to its output?